File: KipSafe/app/src/main/java/com/b18/kipsafe/MainActivity.java
package com.b18.kipsafe;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.ImageButton;
import android.widget.Toast;

import com.b18.kipsafe.Alarms.KipAlarmManager;
import com.b18.kipsafe.Firebase.FirebaseManager;
import com.b18.kipsafe.SunsetCommunication.GetSunSetTask;
import com.firebase.client.Firebase;

public class MainActivity extends AppCompatActivity {

    // The managers used for the main activity.
    private FirebaseManager firebaseManager;
    private KipAlarmManager alarmManager;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        Firebase.setAndroidContext(this);
        setContentView(R.layout.activity_main);

        new UpdateChecker(this).check();

        alarmManager = new KipAlarmManager(this);
//        firebaseManager = new FirebaseManager(alarmManager, this);
//        firebaseManager.setup();

        KipTimeSlider slider = new KipTimeSlider(this);
        slider.setup();

        // Update egg picture, to be sure.
        ImageButton kipButton;
        kipButton = findViewById(R.id.kipButton);
        kipButton.setSelected(alarmManager.isAlarmSet());
    }

    /**
     * fires on button released
     *
     * @param v kip button
     */
    public void hitKip(View v) {
        boolean isAlarmSet = alarmManager.isAlarmSet();
        isAlarmSet = !isAlarmSet;

        // This will also update alarmManager.
//        firebaseManager.changeOpen(isAlarmSet);

        // Provide immediate user feedback
        alarmManager.setIsAlarmSet(isAlarmSet);

        if (isAlarmSet) {
            SharedPreferenceManager sharedPreferenceManager = new SharedPreferenceManager(this);
            // Schedule alarm for the next sunset.
            AlarmSetter handler = new AlarmSetter(this);
            handler.set(sharedPreferenceManager.getPrefTime(), GetSunSetTask.Delay.NO_DELAY);
        } else {
            alarmManager.cancelAlarm();
            Toast.makeText(getBaseContext(), "Alarm canceled", Toast.LENGTH_SHORT).show();
        }
    }

}

File: app/src/main/java/com/b18/kipsafe/MainActivity.kt
package com.b18.kipsafe

import android.os.Bundle
import android.view.View
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import com.b18.kipsafe.alarms.KipAlarmManager
import com.b18.kipsafe.databinding.ActivityMainBinding
import com.b18.kipsafe.firebase.FirebaseManager
import com.b18.kipsafe.sunsetcommunication.GetSunsetTask

class MainActivity : AppCompatActivity() {
    private lateinit var alarmManager: KipAlarmManager
    lateinit var binding: ActivityMainBinding

    private val database = FirebaseManager(this)

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)

        alarmManager = KipAlarmManager(this)
        KipTimeSlider(this).setup()

        // Update the egg picture.
        val kipButton = binding.kipButton
        kipButton.isSelected = alarmManager.isAlarmSet()

        // Make a checkbox to set alarm on weekend only.
        WeekendCheckBox(this)

        database.setupListener()
    }

    /**
     * Fires when the kip button is released.
     *
     * @param view Kip button.
     */
    fun hitKip(view: View) {
        var isAlarmSet = alarmManager.isAlarmSet()
        isAlarmSet = !isAlarmSet

        alarmManager.setIsAlarmSet(isAlarmSet)
        database.updateValue(isAlarmSet)

        if(isAlarmSet) {
            val sharedPreferenceManager = SharedPreferenceManager(this)
            val handler = AlarmSetter(this)
            handler.set(sharedPreferenceManager.minutesBeforeSunset,
                    GetSunsetTask.Delay.NO_DELAY)
        } else {
            alarmManager.cancelAlarm()
            Toast.makeText(baseContext, R.string.cancel_alarm, Toast.LENGTH_SHORT).show()
        }
    }
}